GTG/core/tasks2.py

```diff
@@ -0,0 +1,559 @@
+
+
+    @date_start.setter
+    def date_start(self, value: Any) -> None:
+        self._date_start = Date(value)
+
+
+    @property
+    def date_closed(self) -> Date:
+        return self._date_closed
+
+
+    @date_closed.setter
+    def date_closed(self, value: Any) -> None:
+        self._date_closed = Date(value)
+
+
+    @property
+    def date_modified(self) -> Date:
+        return self._date_modified
+
+
+    @date_modified.setter
+    def date_modified(self, value: Any) -> None:
+        self._date_modified = Date(value)
+
+
+    @property
+    def title(self) -> str:
+        return self.raw_title
+
+
+    @title.setter
+    def title(self, value) -> None:
+        self.raw_title = value.strip('\t\n') or _('(no title)')
+
+
+    @property
+    def excerpt(self) -> str:
+        if not self.content:
+            return ''
+
+        # Strip tags
+        txt = TAG_REGEX.sub('', self.content)
+
+        # Strip subtasks
+        txt = SUB_REGEX.sub('', txt)
+
+        # Strip blank lines and set within char limit
+        return f'{txt.strip()[:80]}…'
+
+
+    def add_tag(self, tag: Tag2) -> None:
+        """Add a tag to this task."""
+
+        if isinstance(tag, Tag2):
+            if tag not in self.tags:
+                self.tags.append(tag)
+
+                for child in self.children:
+                    child.add_tag(tag)
+        else:
+            raise ValueError
+
+
+    def remove_tag(self, tag_name: str) -> None:
+        """Remove a tag from this task."""
+
+        for t in self.tags:
+            if t.name == tag_name:
+                self.tags.remove(t)
+                (self.content.replace(f'{tag_name}\n\n', '')
+                             .replace(f'{tag_name},', '')
+                             .replace(f'{tag_name}', ''))
+
+        for child in self.children:
+            child.remove_tag(tag_name)
+
+    @property
+    def days_left(self) -> Optional[Date]:
+        return self.date_due.days_left()
+
+
+    def update_modified(self) -> None:
+        """Update the modified property."""
+
+        self.modified = Date(datetime.datetime.now())
+
+
+    def __str__(self) -> str:
+        """String representation."""
+
+        return f'Task: {self.title} ({self.id})'
+
+
+    def __repr__(self) -> str:
+        """String representation."""
+
+        tags = ', '.join([t.name for t in self.tags])
+        return (f'Task "{self.title}" with id "{self.id}".'
+                f'Status: {self.status}, tags: {tags}')
+
+
+    def __eq__(self, other) -> bool:
+        """Equivalence."""
+
+        return self.id == other.id
+
+
+    def __hash__(self) -> int:
+        """Hash (used for dicts and sets)."""
+
+        return hash(self.id)
+
+
+# ------------------------------------------------------------------------------
+# STORE
+# ------------------------------------------------------------------------------
+
+class TaskStore(BaseStore):
+    """A list of tasks."""
+
+    __gtype_name__ = 'TaskStore'
```

Comment:
I would include the application, like `gtg_TaskStore` to possibly prevent any (admittedly unlikely) conflicts.

Reply:
I thought these were "local" to the application, changing

Reply:
To my knowledge, the GType name namespace is shared inside an application between all loaded types, but not across processes. So another application using `"TaskStore"` that we don't load wouldn't conflict, but my general rule is to include some namespace just in case, which is why I suggested it.